launcher/src/vscode-launcher.ts

```diff
@@ -66,6 +66,17 @@ export class VSCodeLauncher {
       env.NODE_EXTRA_CA_CERTS = NODE_EXTRA_CERTIFICATE;
     }
 
+    if (!env.SHELL) {
```

Comment:
What about checking if the user shell (in `/etc/passwd`) is /sbin/nologin or not? If cri-o hasn't patched `/etc/passwd` (i.e. containerd is the container runtime) there may be a valid shell for current user anyway.

Beyond that I have tested the PR using image `quay.io/che-incubator-pull-requests/che-code:pr-284-amd64` and the [quarkus devfile from registry.devfile.io](https://registry.devfile.io/devfiles/java-quarkus/1.3.0) and the problem got solved 🎉 . I had an unrelated problem though and [I have filed a separate issue](https://github.com/eclipse/che/issues/22601).

Reply:
This is probably a good idea -- I didn't initially do it since I was trying to keep it simple (since IIRC the default UDI image sets SHELL) but better safe than sorry :)

Reply:
Updated the PR now. If /etc/passwd contains anything other than `/sbin/nologin`, we no longer override it by setting SHELL if it's not already set.